### PR TITLE
Add classname and styles to Voronoi cells

### LINKED ISSUE #684: Changing mouse cursor while using Voronoi
I've made an interactive histogram (using `VerticalRectSeries` and `Voronoi`), and have attached `onHover` and `onClick` handlers to the Voronoi. This is how it currently looks:
without

As can be seen from the gif above, the cursor is the default. I'd like to change it to the `pointer` cursor, when hovering over buckets with data, like:
with

Normally, this is trivial to achieve with CSS but since the Voronoi is positioned on top of the other svg elements, I can't just add `cursor:pointer;` to them.

I've created a small example but for some reason it doesn't run on Codepen (I've experienced this before with react-vis. Not sure if I'm holding it wrong, or there is a "trick"):
https://codepen.io/sqren/pen/vWqPBK?editors=0010

### PR DESCRIPTION
**Pull request by sqren**
Closes #684

This change makes it possible to conditionally add classnames and styles to Voronoi cell, depending on the data points.

This PR also adds initial docs for the Voronoi diagram.

### CODE REVIEW

File: src/plot/voronoi.js [hunk 1]
```diff
@@ -28,7 +29,7 @@ function Voronoi({
     <g className={`${className} rv-voronoi`} style={style}>
       {voronoiInstance.polygons(nodes).map((d, i) => (
         <path
-          className="rv-voronoi__cell"
+          className={`rv-voronoi__cell ${cellClassName(d.data)}`}
```
Comment (sqren) [inline src/plot/voronoi.js]: I'm not a huge fan of the name `cellClassName` and perhaps it should also allow strings (not just functions). But I didn't want to make it more complex.

File: src/plot/voronoi.js [hunk 2]
```diff
@@ -56,7 +57,8 @@ Voronoi.defaultProps = {
   onMouseDown: NOOP,
   onMouseUp: NOOP,
   x: d => d.x,
-  y: d => d.y
+  y: d => d.y,
```
Comment (mcnuttandrew) [inline src/plot/voronoi.js]: might be better to have the default be something like:

```
cellClassName: d => d.className || ''
```

File: src/plot/voronoi.js [hunk 3]
```diff
@@ -71,7 +73,8 @@ Voronoi.propTypes = {
   onMouseUp: PropTypes.func,
   onHover: PropTypes.func,
   x: PropTypes.func,
-  y: PropTypes.func
+  y: PropTypes.func,
+  cellClassName: PropTypes.func
```
Comment (mcnuttandrew) [inline src/plot/voronoi.js]: alphabetize here and above

File: src/plot/voronoi.js (hunk 2 above)
Comment (sqren) [reply src/plot/voronoi.js]: Where would `d.className` come from? I can see that `d` is a polygon node from the voronoi instance but not clear on how classnames are attached to those.

Comment (sqren) [reply src/plot/voronoi.js]: I see now that it comes from `d.data.className`, which is passed to the Voronoi as `nodes`.That makes me think, that `cellClassName` is unnecessary. I'll update this PR.

**Comment by sqren** [comment]
@mcnuttandrew I've updated the code and added a test for the Voronoi diagram. 
Starting from scratch I found it a little hard to add docs (not my strong suit). But I added a base, that will hopefully encourage others to improve upon it. I also wasn't clear on how to test the docs. `npm start` only gives me the show-case items. So there might be some glitches that I didn't catch.

Let me know what you think.

File: src/plot/voronoi.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ function Voronoi({
     <g className={`${className} rv-voronoi`} style={style}>
       {voronoiInstance.polygons(nodes).map((d, i) => (
         <path
-          className="rv-voronoi__cell"
+          className={`rv-voronoi__cell ${d.data.className || ''}`}
```
Comment (mcnuttandrew) [inline src/plot/voronoi.js]: just for paranoia maybe do d.data && d.data.className

**Comment by sqren** [comment]
@mcnuttandrew Good to hear :) Commited the last fixes.

**Comment by sqren** [comment]
@mcnuttandrew Any idea when this will be published on npm? Thanks!

**Comment by mcnuttandrew** [comment]
@balthazar ?

**Comment by balthazar** [comment]
`1.8.1`

**Pull request merged**